libs/cart/src/drivers/magento/cart-item.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable, Inject } from '@angular/core';
+import { Apollo } from 'apollo-angular';
+
+import { Observable } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+
+import {
+  DaffProductServiceInterface,
+  DaffProductUnion,
+  DaffProductDriver
+} from '@daffodil/product'
+
+import { DaffCartItemServiceInterface } from '../interfaces/cart-item-service.interface';
+import { DaffCart } from '../../models/cart';
+import { DaffCartItem } from '../../models/cart-item';
+import { DaffCartItemInput } from '../../models/cart-item-input';
+import { DaffMagentoCartTransformer } from './transforms/outputs/cart.service';
+import {
+  listCartItems,
+  addCartItem,
+  removeCartItem,
+  updateCartItem
+} from './queries';
+import { DaffMagentoCartItemTransformer } from './transforms/outputs/cart-item.service';
+import { DaffMagentoCartItemInputTransformer } from './transforms/inputs/cart-item.service';
+import { MagentoListCartItemsResponse } from './models/responses/list-cart-items';
+import { MagentoAddCartItemResponse } from './models/responses/add-cart-item';
+import { MagentoRemoveCartItemResponse } from './models/responses/remove-cart-item';
+import { DaffMagentoCartItemUpdateInputTransformer } from './transforms/inputs/cart-item-update.service';
+import { MagentoUpdateCartItemResponse } from './models/responses';
+
+/**
+ * A service for making Magento GraphQL queries for carts.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartItemService implements DaffCartItemServiceInterface<
+  DaffCartItem,
+  DaffCartItemInput,
+  DaffCart
+> {
+  constructor(
+    private apollo: Apollo,
+    @Inject(DaffProductDriver) private product: DaffProductServiceInterface<DaffProductUnion>,
+    public cartTransformer: DaffMagentoCartTransformer,
+    public cartItemTransformer: DaffMagentoCartItemTransformer,
+    public cartItemInputTransformer: DaffMagentoCartItemInputTransformer,
+    public cartItemUpdateInputTransformer: DaffMagentoCartItemUpdateInputTransformer
+  ) {}
+
+  list(cartId: string): Observable<DaffCartItem[]> {
+    return this.apollo.query<MagentoListCartItemsResponse>({
+      query: listCartItems,
+      variables: {cartId}
+    }).pipe(
+      map(result =>
+        result.data.cart.items.map(item => this.cartItemTransformer.transform(item))
+      )
+    )
+  }
+
+  get(cartId: string, itemId: number): Observable<DaffCartItem> {
+    return this.list(cartId).pipe(
+      map(items => items.find(item => Number(item.item_id) === itemId))
+    )
+  }
+
+  add(cartId: string, product: DaffCartItemInput): Observable<Partial<DaffCart>> {
+    return this.product.get(product.productId).pipe(
```

Comment:
I think I understand why you did this. Because you needed the sku of the product? But with magento, we treat the product.id as the same as the sku (because you can't actually filter for products on the magentoProduct.id. You need to use the magentoProduct.sku. Apparently you can't add a product to the cart by magentoProduct.id either). So you can just take in the product id from the user, and just transform it back into the sku. Then you can get rid of that MagentoDaffCartInputWithSku object too.

Reply:
Ah well that will certainly be simpler. Changing now.

Reply:
Changed in [`931e356`](https://github.com/graycoreio/daffodil/pull/676/commits/931e356340d5188dea7fc0865152377868fef06b).